### PR TITLE
Video tutorials moved to a new dialog

### PR DESCRIPTION
**Pull request by xavijam**
- New place to open video tutorials.

screen shot 2015-08-20 at 14 22 57

---

- Video tutorials list.

screen shot 2015-08-20 at 13 58 24

---

- Video preview.

screen shot 2015-08-20 at 13 58 18

---

This change could affect to:
- Docs and the mentioned video tutorials. @csobier @brittanymicek @iriberri 
- Templates workflow: we should merge master when we have time, but it will change several things for this feature. @javisantana @alonsogarciapablo 

REVIEWER: @viddo 
@josecruz, let's review the UI together :).

### CODE REVIEW

**Comment by saleiva** [comment]
:_)

**Comment by xavijam** [comment]
Could you take care @viddo? Lot of files, but several deletions, you know. It will available in rui servers in a moment.

**Comment by xavijam** [comment]
retest this please

File: lib/assets/javascripts/cartodb/common/dialogs/video_tutorial/video_tutorial_view.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      videoId: this.options.videoId
+    });
+    this.elder('initialize');
+    this.template = cdb.templates.getTemplate('common/dialogs/video_tutorial/video_tutorial_template');
+    this._initBinds();
+  },
+
+  render: function() {
+    BaseDialog.prototype.render.call(this);
+    this._initViews();
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/common/dialogs/video_tutorial/video_tutorial_view.js]: Considering overriding the render call and adding subviews on the render, shouldn't there be a `this.clearSubViews()` above here?

Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/video_tutorial/video_tutorial_view.js]: Agree

File: lib/assets/test/spec/cartodb/dashboard/dialogs/video_tutorial.spec.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  });
+
+  describe('list', function() {
+    it('should render items properly', function() {
+      expect(this.view.$('.VideoTutorial-item').length).toBe(6);
+    });
+
+    it('should trigger template selected when item is clicked', function() {
+      var eventTriggered = false;
+      cdb.god.bind('onTemplateSelected', function() {
```
Comment (viddo) [inline lib/assets/test/spec/cartodb/dashboard/dialogs/video_tutorial.spec.js]: if you want to assert a callback is called you can make it slightly simpler by just using a spy, i.e.

```js
var onTemplateSelectedSpy = jasmine.createSpy('onTemplateSelected');
cdb.god.bind('onTemplateSelected', onTemplateSelectedSpy);
expect(onTemplateSelectedSpy).toHaveBeenCalled()
```

File: lib/assets/javascripts/cartodb/dashboard/entry.js [hunk 3]
```diff
@@ -121,6 +122,17 @@ $(function() {
...
       });
 
+      createModel.bind('startTutorial', function() {
+        createDialog.close();
+        
+        var dlg = new VideoTutorialView({
+          clean_on_hide: true,
+          enter_to_confirm: false
+        })
+        cdb.god.trigger("onTemplateSelected", this);
```
Comment (viddo) [inline lib/assets/javascripts/cartodb/dashboard/entry.js]: Is this event really necessary to be triggered?

**Comment by viddo** [comment]
Code wise just some minor details, all :+1:

File: lib/assets/javascripts/cartodb/dashboard/entry.js (hunk 3 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb/dashboard/entry.js]: Seems like, it is related to open/close video player (cc @javierarce). It is code from the previous behaviour, nothing new.

Comment (javierarce) [inline lib/assets/javascripts/cartodb/dashboard/entry.js]: Yes, it's related to the tutorials.

File: lib/assets/test/spec/cartodb/dashboard/dialogs/video_tutorial.spec.js (hunk 2 above)
Comment (xavijam) [inline lib/assets/test/spec/cartodb/dashboard/dialogs/video_tutorial.spec.js]: Thanks man! :)

**Pull request merged**